### ISSUE TITLE
update-initramfs issues on ALL debian-based distros

### ISSUE THREAD

**Issue by snazzybunny** [cawilliamson/ansible-gpdpocket#82]
I must enter username in the tty after I selected install ubuntu in grub. A graphical interface does not load up and I tried it on two different usb sticks.

**Comment by cawilliamson** [comment]
I'll look in to it when I'm feeling a bit more up to it.

I've been working on this stuff all weekend so far. Need a break.

**Comment by snazzybunny** [comment]
No problem dude. Go enjoy the weekend :)/

**Comment by nexus511** [comment]
I basically can also look into this as I need to verify some things on ubuntu 16.04.3 anyways today. But it will not happen during the next 4-6 hours.

**Comment by nexus511** [comment]
Okay. Neither the installer, nor the live-system are coming up again on
- ubuntu 16.04.3
- xubuntu 16.04.3
anymore.

X is basically available and functional on those systems (startx brings up an X session). So the login managers seem to be broken for installer and ubiquity.

**Comment by nexus511** [comment]
For some reason it seems like /usr/sbin/lightdm is now missing. This is used as display manager for ubiquity and the live session though.

**Comment by nexus511** [comment]
Script outputs

```
TASK [boot : setup boot environment for system (iso creation = no)] ************
```

while iso is build. So I guess that the chroot detection does not work here.

**Comment by nexus511** [comment]
Okay, there we are. Removing initramfs-tools is a really bad idea though, as those kick out lightdm out as well:

```
changed: [localhost] => (item={u'state': u'absent', u'name': u'initramfs-tools'}) => {"changed": true, "item": {"name": "initramfs-tools", "state": "absent"}, 
"stderr": "E: Can not write log (Is /dev/pts mounted?) - posix_openpt (2: No such file or directory)
dpkg: plymouth: dependency problems, but removing anyway as you requested:
 lightdm depends on plymouth (>= 0.8.8-0ubuntu18).
 plymouth-label depends on plymouth (= 0.9.2-3ubuntu13.1).
 mountall depends on plymouth.

dpkg: initramfs-tools: dependency problems, but removing anyway as you requested:
 ubuntu-minimal depends on initramfs-tools.
 linux-image-4.10.0-28-generic depends on initramfs-tools | linux-initramfs-tool; however:
  Package initramfs-tools is to be removed.
  Package linux-initramfs-tool is not installed.
  Package initramfs-tools which provides linux-initramfs-tool is to be removed.
 apparmor depends on initramfs-tools | linux-initramfs-tool; however:
  Package initramfs-tools is to be removed.
  Package linux-initramfs-tool is not installed.
  Package initramfs-tools which provides linux-initramfs-tool is to be removed.
 console-setup depends on initramfs-tools (>= 0.85eubuntu12) | linux-initramfs-tool; however:
  Package initramfs-tools is to be removed.
  Package linux-initramfs-tool is not installed.
  Package initramfs-tools which provides linux-initramfs-tool is to be removed.
 linux-image-4.10.0-28-generic depends on initramfs-tools | linux-initramfs-tool; however:
  Package initramfs-tools is to be rem
dpkg: error: --compare-versions takes three arguments: <version> <relation> <version>

Type dpkg --help for help about installing and deinstalling packages [*];
Use 'apt' or 'aptitude' for user-friendly package management;
Type dpkg -Dhelp for a list of dpkg debug flag values;
Type dpkg --force-help for a list of forcing options;
Type dpkg-deb --help for help about manipulating *.deb files;

Options marked [*] produce a lot of output - pipe it through 'less' or 'more' !
dracut: caps: does not work with systemd in the initramfs
```

**Comment by RevoltsRevolution** [comment]
Hey there... I'm the guy from ticket #63 - tried the newest release and get a failure while trying. I think it's not updating grub, at least it echos: "I can't update grub config - this is bad!" - is this sure the right spot to discuss this, since Ubuntu is not Kali? I know both are Debian based, but I also think that Kali is a tad away from Ubuntu... :0

**Comment by nexus511** [comment]
@RevoltsRevolution No, theese are two things. When this is in some way related to one of the Ubuntu tickets, it is #85.

As #85 is related to ubuntu, as the title clearly states, you might want to create a new ticket after you really tried the latest changes on the tree and are sure that you have verified that the fix for #85 does not work for Kali at all.

**Comment by cawilliamson** [comment]
Not sure why this one got closed - sorry all!

**Comment by RevoltsRevolution** [comment]
I tried the latest version and it sadly fails. The log is so impossible big that I'm not sure I should post it. I don't know what to do anymore :/ tried every version but it fails either to make the iso or it fails to boot at all. I don't know why Kali could boot on your GPD cawilliamson :/

**Comment by cawilliamson** [comment]
@RevoltsRevolution I've added a fix for the failing ISO creation now.

Can you build a new ISO, boot it and let me know if the problem still remains? Don't be disheartened if it does - I suspect it may well do and will work on fixing that but one step at a time. :)

**Comment by cawilliamson** [comment]
I have added a fix for this on ticket #102 

I'm closing this one in favour of that since they're actually ultimately about the same problem.

Please test the fix I've added and let me know if the issue is resolved - if so, thank @psrb191921 ! 👍
